qiskit/compiler/transpiler.py

```diff
@@ -339,10 +338,7 @@ def callback_func(**kwargs):
         if translation_method is None and hasattr(backend, "get_translation_stage_plugin"):
             translation_method = backend.get_translation_stage_plugin()
 
-    initial_layout = _parse_initial_layout(initial_layout)
-    approximation_degree = _parse_approximation_degree(approximation_degree)
```

Comment:
before this PR, setting a custom `initial_layout` or `approximation_degree` could lead to different results on `generate_preset_pass_manager` vs `transpile`, while the rest of inputs would be consistent.